src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs

```diff
@@ -156,7 +156,7 @@ public Task ConnectAsync(Uri uri, CancellationToken cancellationToken)
 
         private Task ConnectAsyncCore(Uri uri, CancellationToken cancellationToken)
         {
-            _innerWebSocket = new WinHttpWebSocket();
+            _innerWebSocket = WebSocketHandle.Create();
 
```

Comment:
This isn't a sufficient abstraction for handling multiple platforms as the exception handling is insufficient for NetNative. The ConnectAsyncCore implementation needs to be pushed into platform specific code.